lib/spack/spack/util/spack_yaml.py

```diff
@@ -32,8 +32,7 @@
 
 
 # Make new classes so we can add custom attributes.
-# Also, use OrderedDict instead of just dict.
-class syaml_dict(collections.OrderedDict):
+class syaml_dict(dict):
```

Comment:
Can we get rid of the custom `__repr__` too?

Reply:
fwiw, i've also tried adding __slots__ for the custom attributes, but that didn't help perf wise. i could add it back through as its a form of documentation